.github/workflows/cpp.yml

```diff
@@ -194,6 +194,8 @@ jobs:
       ARROW_BUILD_TESTS: ON
       ARROW_DATASET: ON
       ARROW_FLIGHT: ON
+      ARROW_FLIGHT_SQL: ON
+      ARROW_FLIGHT_SQL_ODBC: ON
       ARROW_GANDIVA: ON
```

Comment:
Good job Justin! You did it! Seems the ODBC can be built on Intel macOS too in your PR